soda_machine.rb

```diff
@@ -1,18 +1,32 @@
 class SodaMachine
-  attr_reader :sodas, :cash
+  attr_reader :sodas, :cash, :inventory
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
   end
 
   def current_inventory_count
+    return @sodas.size
   end
 
   def find_soda(soda_brand)
+    @sodas.each do |soda|
+      if soda.brand == soda_brand
+        return soda
+      else
+        return nil
+      end
+    end
   end
+# Got caught up with the previous method and ran out of time!
 
   def sell(soda_brand)
+    @sodas.each do |soda|
```

Comment:
How could you adjust this to ensure that the `Soda` instance of concern gets removed from the `@sodas` array? Also, how could this be refactored to leverage the `find_soda` method already implemented.